msteams-platform/concepts/deploy-and-publish/appsource/prepare/frequently-failed-cases.md

```diff
@@ -1,246 +1,322 @@
+* Must display content in a single column and not have multiple navigation levels.
+* Must not use task modules.
+* Should align with the center of the meeting stage.
+* Should be dismissed once a user selects a button or performs an action.
 
-### &#9989; Do not post sensitive data to an audience not intended to view the data
+## 10.0 Notification design
 
-Your Teams app must not post sensitive data such as credit card or financial payment instrument, Personal Identifiable Information (PIN), health, or contact tracing information to an audience not intended to view that data.
+Follow these guidelines if your app uses the [activity feed APIs provided by Microsoft Graph](https://docs.microsoft.com/graph/teams-send-activityfeednotifications).
 
-### &#9989; Do not transmit financial payment details or complete financial transactions via your Teams app
+### 10.1 General
 
-* Your Teams app must not ask users to make a payment directly within Teams interface.
-* Apps may not transmit financial instrument details through the user on the app interface. Apps may only transmit links to secure payment services to users if this is disclosed in the app's Terms of Use, Privacy Policy, and any profile page or website for the app before a user agrees to use the app.
+* All the notification triggers specified in your app manifest should get a notification in the app.
+* The notifications must be localized as per the supported languages declared in the app manifest.
+* A notification must load within 5 seconds of a user action.
 
-### &#9989; Clear warning before downloading any files or executable (`.exe`) into a user’s environment
+### 10.2 Avatars
 
-Please warn users before your app downloads any files or executable (`.exe`  )into the user's machine or environment.
+* The notification avatar should match the app icon provided in the manifest.
+* Notifications triggered by an app user should contain the avatar of the user who initiated it.
 
-### &#9989; Messaging extensions must provide help text and be easy to read
+### 10.3 Spamming
 
-* The search-based messaging extension must provide help text on how to effectively search (e.g., show example input).
-* Task modules must include an icon and a short name that they are contained in or created from the app.
-* The message extension `@mention` executables must be clear, easy to understand, and easy to read.
-![Message extension](../../../../assets/images/faq/message-extension.png)
+* Apps must not send more than 10 notifications per minute to an individual user.
+* Bots and the activity feed should not trigger duplicate notifications.
+* Notifications must provide some value to users and not be used for trivial or irrelevant events.
 
-## M365 Publisher Attestation
+### 10.4 Navigation and layout
 
-### &#9989; Complete the Publisher Attestation in Partner Center
+* The right pane experience in the feed must be self-contained and not break the feed experience.
```

Comment:
What is the "right pane experience in the feed" referring to? The chat pane?

Reply:
Notification pane

Reply:
Thanks! I am still a little unclear about what "must be self-contained and not break the feed experience". Can you elaborate, @ramendhe?